src/main/java/world/bentobox/bentobox/listeners/BlockEndDragon.java

```diff
@@ -23,7 +24,7 @@ public BlockEndDragon(BentoBox plugin) {
      */
     @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
     public boolean onDragonSpawn(CreatureSpawnEvent e) {
-        if (!e.getEntityType().equals(EntityType.ENDER_DRAGON) || plugin.getIWM().isDragonSpawn(e.getEntity().getWorld())) {
+        if (!e.getEntityType().equals(EntityType.ENDER_DRAGON) || plugin.getWorlds().getGameWorld(e.getEntity().getWorld()).map(gameWorld -> gameWorld.getSettings().isDragonSpawn()).orElse(false)) {
```

Comment:
I think this is a bit "too much" there. How could we make this API call less big?

Reply:
One way is to create new method in BentoBox plugin
`plugin.getGameWorld(World) { return this.worldManager.getGameWorld(World); } `

Another good nanosecond optimization is to create new variable in this method
`if (!e.getEntityType().equals(EntityType.ENDER_DRAGON))`
`{`
`World world = e.getEntity().getWorld();`
`    if (plugin.getGameWorld(world).map(gameWorld -> gameWorld.getSettings().isDragonSpawn()).orElse(false)))
    {
        // do staff
    }`

Reply:
I'm not sure if the `plugin.getGameWorld(World) { return this.worldManager.getGameWorld(World); }`  method would be a good idea.

In regards of the variable - that's a no, I think. That'd definitely make the condition look "lighter" but it'll mainly create this unnecessary variable at each call.

Reply:
So internal variable is an optimization, as it avoid multiple calls for **.getEntity().getWorld()**
Its small, but in some cases a necessary optimization, to avoid unnecessary method calls.

Reply:
This is not an optimization. `getEntity().getWorld()` are just getters that return a value that is part of the object - there is no calculations done there.
What I could do, however, is providing a method to directly access the GameSettings of a GameWorld...